core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -95,15 +95,21 @@ private DataTask task(TableScan scan) {
     }
   }
 
-  static class PartitionSet {
+  static class PartitionMap {
```

Comment:
This was renamed to avoid a conflict with the new `PartitionSet` class, and updated to use `StructLikeWrapper` with correct `equals`/`hashCode` implementations.